docs/source/concept_guides/quantization.mdx

```diff
@@ -1,5 +1,256 @@
+  - 11 bits in `float64`
+2. The mantissa
+  - 11 bits in `float16` (10 explictly stored)
+  - 8 bits in `bfloat16` (7 explicitly stored)
+  - 24 bits in `float32` (23 explictly stored)
+  - 53 bits in `float64` (52 explictly stored)
+
+For more information on the bits allocation for each data type, check the nice illustration on the Wikipedia page about
+the [bfloat16 floating-point format](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format).
+
+For a real number `x` we have:
+
+```
+x = sign x mantissa x (2^exponent)
+```
+
+### Quantization
+
+The two most common quantization cases are `float32 -> float16` and `float32 -> int8`.
+
+#### Quantization to `float16`
+
+Performing quantization to go from `float32` to `float16` is quite straightforward since both data types follow the same
+representation scheme. The questions to ask yourself when quantizing an operation to `float16` are:
+
+- Does my operation have a `float16` implementation?
+- Does my hardware suport `float16`? For instance, Intel CPUs [have been supporting `float16` as a storage type, but
+computation is done after converting to `float32`](https://scicomp.stackexchange.com/a/35193). Full support will come
+in Cooper Lake and Sapphire Rapids.
+- Is my operation sensitive to lower precision?
+For instance the value of epsilon in `LayerNorm` is usually very small (~ `1e-12`), but the smallest representable value in
+`float16` is ~ `6e-5`, this can cause `NaN` issues.  The same applies for big values.
+
+#### Quantization to `int8`
+
+Performing quantization to go from `float32` to `int8` is more tricky. Only 256 values can be represented in `int8`,
+while `float32` can represent a very wide range of values. The idea is to find the best way to project our range `[a, b]`
+of `float32` values  to the `int8` space.
+
+Let's consider a float `x` in `[a, b]`, then we can write the following quantization scheme, also called the *affine
+quantization scheme*:
+
+```
+x = S * (x_q - Z)
+```
+
+where:
+
+- `x_q` is the quantized `int8` value associated to `x`
+- `S` and `Z` are the quantization parameters
+  - `S` is the scale, and is a positive `float32`
+  - `Z` is called the zero-point, it is the `int8` value corresponding to the value `0` in the `float32` realm. This is
+  important to be able to represent exactly the value `0` because it is used everywhere throughout machine learning
+  models.
+
+
+The quantized value `x_q` of `x` in `[a, b]` can be computed as follows:
+
+```
+x_q = round(x/S + Z)
+```
+
+And `float32` values outside of the `[a, b]` range are clipped to the closest representable value, so for any
+floating-point number `x`:
+
+```
+x_q = clip(round(x/S + Z), round(a/S + Z), round(b/S + Z))
+
+```
+
+This is called the *affine quantization sheme* because the mapping from `[a, b]` to `int8` is an affine one.
+
+A common special case of this scheme is the *scale quantization scheme*, where we consider a symmetric range of float values `[-a, a]`.
```

Comment:
I never read this term. To me what this paragraph describes is symmetric vs asymmetric no?